### PR TITLE
Fix jvm-*.options when upgrade Docker Cassandra 3 -> 4...

### PR DESCRIPTION
**Pull request by vttranlina**
Time  to start up the Cassandra container on my pc 

Before: 67 seconds
After: 13 seconds

### CODE REVIEW

**Review by Arsnael — APPROVED** [review]

**Review by quantranhong1999 — APPROVED** [review]
Good catch! Hope we would get better stable build.

**Review by chibenwa — APPROVED** [review]

**Review by hungphan227 — APPROVED** [review]

**Comment by chibenwa** [comment]
`org.apache.james.backends.cassandra.init.SessionWithInitializedTablesFactoryTest.createSessionShouldKeepTheSetSchemaVersionWhenTypesAndTablesHaveNotChanged`

`Query timed out after PT2S`

Maybe give it a bit more memory no?

**Comment by vttranlina** [comment]
CI still fail
```
com.datastax.oss.driver.api.core.DriverTimeoutException: Query timed out after PT2S
getAllMappingsShouldReturnMultipleStoredMappings – org.apache.james.rrt.cassandra.CassandraRecipientRewriteTableDAOTest

readBytesShouldReturnBigSavedByteSource(StoragePolicy)[3] – org.apache.james.blob.cassandra.CassandraBlobStoreClOneTest
```

I will commit no limit memory, and see what happens, (In the past when miss jvm.**options, the pr was green)
That will help trace the reason comes from in-memory exceed or not

**Comment by vttranlina** [comment]
Ci green, 
When we don't provide configured the Xms, Xmx, 
the `cassandra-env.sh` algrothim
```
    # set max heap size based on the following
    # max(min(1/2 ram, 1024MB), min(1/4 ram, 8GB))
```
=> CI server has 48GB => Cassandra container will run with 8GB

_______
Let's try with 2GB

**Comment by Arsnael** [comment]
Sure it's alright :) I remember back then we put a limit on those containers so people could run tests locally without bloating their RAM, but for the CI probably better to use more

**Comment by quantranhong1999** [comment]
> but for the CI probably better to use more

Maybe we could set an env variable for CI env acknowledge, and default to local env?
rf: https://github.com/apache/james-project/blob/master/Jenkinsfile#L39

File: backends-common/cassandra/src/test/java/org/apache/james/backends/cassandra/CassandraCluster.java [hunk 1]
```diff
@@ -64,7 +66,12 @@ private CassandraCluster(CassandraModule module, Host host) throws RuntimeExcept
             .keyspace(KEYSPACE)
             .replicationFactor(1)
             .disableDurableWrites();
-        this.nonPrivilegedCluster = ClusterFactory.create(clusterConfiguration, keyspaceConfiguration);
+
+        DriverConfigLoader driverConfigLoader = DriverConfigLoader.programmaticBuilder()
+            .withString(DefaultDriverOption.SESSION_LEAK_THRESHOLD, "16")
```
Comment (Arsnael) [inline backends-common/cassandra/src/test/java/org/apache/james/backends/cassandra/CassandraCluster.java:71]: Why do we have that many session actives and does it affect perf/stability? I find a bit easy to just silence it?

Comment (vttranlina) [reply backends-common/cassandra/src/test/java/org/apache/james/backends/cassandra/CassandraCluster.java:71]: > Why do we have that many session actives and does it affect perf/stability

I find a test creates a new session for each test and not close to it, I think we have more similarities somewhere 

When I read a log from `test-run.log` of the fail test case, I see a suspect warning
```

21:08:37.630 [WARN ] c.d.o.d.i.c.session.DefaultSession - You have too many session instances: 10 active, expected less than 4 (see 'advanced.session-leak.threshold' in the configuration)
21:08:37.981 [WARN ] c.d.o.d.i.c.session.DefaultSession - You have too many session instances: 11 active, expected less than 4 (see 'advanced.session-leak.threshold' in the configuration)
```

**Comment by vttranlina** [comment]
CI fail 
```
targetFolderNamesShouldWorkWhenTargetFolderDoNotExist(File) – org.apache.james.mailets.WithStorageDirectiveIntegrationTest
1m 3s
Error
Condition with lambda expression in org.apache.james.utils.TestIMAPClient was not fulfilled within 1 minutes.
Stacktrace
org.awaitility.core.ConditionTimeoutException: Condition with lambda expression in org.apache.james.utils.TestIMAPClient was not fulfilled within 1 minutes.
```
it looks not related to this pr, can you re-run the ci?

File: backends-common/cassandra/src/test/java/org/apache/james/backends/cassandra/CassandraCluster.java (hunk 1 above)
Comment (vttranlina) [reply backends-common/cassandra/src/test/java/org/apache/james/backends/cassandra/CassandraCluster.java]: CI still fails, It looks not resolve the unstable -> Reverted

File: backends-common/cassandra/src/test/java/org/apache/james/backends/cassandra/DockerCassandra.java [hunk 2]
```diff
@@ -169,14 +169,21 @@ public void onComplete() {
             public void close() {
             }
         });
+
+        String memorySettingCommand = Optional.ofNullable(System.getenv("CI"))
```
Comment (Arsnael) [inline backends-common/cassandra/src/test/java/org/apache/james/backends/cassandra/DockerCassandra.java:173]: Shouldn't we add this somewhere for the CI then? Like maybe the Jenkinsfile?

Comment (vttranlina) [reply backends-common/cassandra/src/test/java/org/apache/james/backends/cassandra/DockerCassandra.java:173]: Your mean is `CI` ?
It is already in Jenkinsfile https://github.com/apache/james-project/pull/1689#issuecomment-1687793011

Comment (Arsnael) [reply backends-common/cassandra/src/test/java/org/apache/james/backends/cassandra/DockerCassandra.java:173]: My bad :)

**Comment by vttranlina** [comment]
green !!!!, I will squash it

**Comment by chibenwa** [comment]
:green_apple: again... Merging!

**Pull request merged**